### PR TITLE
feat: splitting of ambianic-PWA functions into a separate repository

### PR DESCRIPTION
**Pull request by vickywane**
This PR contains the basic serverless functions extracted from this [PR](https://github.com/ambianic/ambianic-ui/pull/562) and managed using Netlify functions for the user-management feature on the PWA.

### CODE REVIEW

**Comment by ivelin** [comment]
@vickywane where is the CI pipeline with local tests and staging tests before pushing live to production?

File: .github/lighthouse/budget.json [hunk 1]
```diff
@@ -0,0 +1,49 @@
+[
+  {
```
Comment (ivelin) [inline .github/lighthouse/budget.json]: @vickywane why do we need a lighthouse config for serverless functions? LH is for PWA apps.

File: .github/workflows/lighthouse.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
+# This is a basic workflow to help you get started with Actions
```
Comment (ivelin) [inline .github/workflows/lighthouse.yml]: @vickywane same point here, LH is not intended for serverless function testing.

File: .github/workflows/nodejs.yml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        node-version: ${{ matrix.node-version }}
+
+    - name: Install npm
+      run: |
+        npm version
+        npm ci
+      env:
+        CI: true
+
+    - name: Test Suite
```
Comment (ivelin) [inline .github/workflows/nodejs.yml]: @vickywane why do we need this workflow? Its intended for PWA testing, not serverless functions.

File: README.md [hunk 4]
```diff
@@ -1,2 +1,4 @@
 # ambianic-subscriptions
 Repo for ambianic premium subscription management
+
+This repo hosts [Netlify functions](https://www.netlify.com/products/functions/) that handle Ambianic premium user subscriptions. This is code that cannot run in the browser PWA, because it needs access keys to subscription and payment gateway services (Stripe) shared between all app users. If the code runs in the PWA, users will be able to see and manipulate each other's premium subscription data.
```
Comment (ivelin) [inline README.md]: @vickywane can you please insert a high level architecture diagram that illustrates roughly the intended interactions between ambianic PWA, ambianic serverless functions (this repo) hosted by netlify(aws), Auth0, stripe and ambianic edge.

Comment (ivelin) [reply README.md]: Here is a template to start from: https://app.diagrams.net/#G1zj8sg3aHGFKSvCbhcQ5BHEmWU4GAC-6X

File: index.html [hunk 5]
```diff
@@ -0,0 +1,21 @@
+<!doctype html>
```
Comment (ivelin) [inline index.html:1]: @vickywane why do we need to serve an html page?

**Review by ivelin — CHANGES_REQUESTED** [review]
@vickywane can we stick with npm for now for uniformity? 

Also, I don't see any postman test scripts in this PR.

See a few other comments inline.

File: index.html (hunk 5 above)
Comment (vickywane) [reply index.html:1]: The page is what is built by Netlify on each deployment. I tried deploying a project without a build file but it breaks the deployment.

File: .github/workflows/nodejs.yml (hunk 3 above)
Comment (vickywane) [reply .github/workflows/nodejs.yml]: I'll remove it.

File: .github/workflows/lighthouse.yml (hunk 2 above)
Comment (vickywane) [reply .github/workflows/lighthouse.yml]: I'll remove it

File: .github/lighthouse/budget.json (hunk 1 above)
Comment (vickywane) [reply .github/lighthouse/budget.json]: I'll remove it

**Comment by vickywane** [comment]
> Also, I don't see any postman test scripts in this PR.

The test scripts are the JSON collection file at `tests/postman/*.json`

**Comment by ivelin** [comment]
@vickywane I don't see any CI running on your PR originating fork:
https://github.com/vickywane/ambianic-subscriptions

As mentioned before, your fork should have a full CI checklist that's at least at the level of or better than ambianic-ui a with 100% pass rate. Once that's accomplished, we can merge it in the main branch.

**Comment by ivelin** [comment]
@vickywane is this PR duplicate of #4 which I just merged?

**Pull request merged**